peachjam/models/core_document_model.py

```diff
@@ -499,10 +499,14 @@ def pre_save(self):
             self.work.title = self.title
             self.work.save()
 
-    def save(self, *args, **kwargs):
+    def save(self, *args, is_extracting_citations=False, **kwargs):
         # give ourselves and subclasses a chance to pre-populate derived fields before saving, in case full_clean() has
         # not yet been called
         self.pre_save()
+
+        # we use this flag in the re_extract_citations signal to prevent an infinite loop
+        self.is_extracting_citations = is_extracting_citations
```

Comment:
@nickmwangemi I managed to fix the infinite loop by adding this flag. I call it when saving after extracting citations and check for it in the signal.

Reply:
Hmm, good point. Perhaps there's a cleaner way...